javascript/modules/client/scripts/generate-examples-md.js

```diff
@@ -0,0 +1,105 @@
+const fs = require("fs");
+const { argv } = require("process");
+const regex = /(\/\* MARKDOWN)(.|\n)*?(\*\/\n)/g;
+const removeStrings = ["/* MARKDOWN\n", "*/"];
+
+generateExamplesMd();
+
+async function generateExamplesMd() {
+  const path = argv[2];
+  const outputFile = argv[3];
+  if (!path || !outputFile) {
+    console.error(
+      "Usage:  node generate-examples-md.js <examples-folder> <output-file.md>"
+    );
+    process.exit(1);
+  }
+  const files = getFileNames(path);
+  if (fs.existsSync(outputFile)) {
+    await fs.promises.rm(outputFile);
+  }
+  for (let i = 0; i < files.length; i++) {
+    const file = files[i];
+    fs.readFile(file, async (err, data) => {
```

Comment:
This is a problem, because the `for` loop and the `readFile` functions are synchronous, while the inner callback is async. We could be continuing the execution flow before `appendFile` has been processed in all callbacks. 

I think we should build an array of promises and then do `Promise.all()`  on it.

Reply:
Not sure if this will break it in the same way as the `for of` loop